torchvision/datasets/folder.py

```diff
@@ -28,11 +28,12 @@ def make_dataset(dir, class_to_idx):
         if not os.path.isdir(d):
             continue
 
-        for filename in os.listdir(d):
-            if is_image_file(filename):
-                path = '{0}/{1}'.format(target, filename)
-                item = (path, class_to_idx[target])
-                images.append(item)
+        for root, _, fnames in sorted(os.walk(d)):
```

Comment:
you definitely dont. `os.listdir` will sometimes return the folders in a random order in linux or python2.7 i think? . I was just keeping with the existing code where it's done for the classes:

```python
def find_classes(dir):
    classes = [d for d in os.listdir(dir) if os.path.isdir(os.path.join(dir, d))]
    classes.sort()
    class_to_idx = {classes[i]: i for i in range(len(classes))}
    return classes, class_to_idx
```